src/AppInstallerCommonCore/Public/AppInstallerProgress.h

```diff
@@ -70,6 +70,9 @@ namespace AppInstaller
 
         // Sets a cancellation function that will be called when the operation is to be cancelled.
         [[nodiscard]] virtual CancelFunctionRemoval SetCancellationFunction(std::function<void()>&& f) = 0;
+
+        // Waits with cancellation. Returns false if cancelled before timeout.
+        virtual bool Wait(std::chrono::milliseconds millisecondsToWait) = 0;

```

Comment:
I meant a standalone helper function, or a static if you prefer.  It definitely doesn't need to be on the interface itself.